cpp/tests/experimental/louvain_primitives.cu

```diff
@@ -0,0 +1,197 @@
+#include <experimental/graph.hpp>
+
+#include <patterns/copy_v_transform_reduce_key_aggregated_out_nbr.cuh>
+#include <patterns/transform_reduce_by_adj_matrix_row_col_key_e.cuh>
+#include <utilities/dataframe_buffer.cuh>
+#include <utilities/shuffle_comm.cuh>
+
+#include <rmm/device_uvector.hpp>
+
+template <typename graph_t>
+void update_by_delta_modularity(raft::handle_t const& handle,
+                                graph_t const& graph_view,
+                                std::vector<typename graph_t::vertex_type> cluster_h,
+                                typename graph_t::weight_type total_edge_weight,
+                                typename graph_t::weight_type resolution)
+{
+  using vertex_t = typename graph_t::vertex_type;
+  using edge_t   = typename graph_t::edge_type;
+  using weight_t = typename graph_t::weight_type;
+
+  vertex_t num_verts = graph_view.get_number_of_vertices();
+
+  rmm::device_uvector<vertex_t> cluster_v(num_verts, handle.get_stream());
+
+  raft::update_device(cluster_v.data(), cluster_h.data(), num_verts, handle.get_stream());
+
+  rmm::device_uvector<vertex_t> communities(0, handle.get_stream());
+  rmm::device_uvector<weight_t> community_in_weight_sums(0, handle.get_stream());
+
+  std::tie(communities, community_in_weight_sums) =
+    cugraph::experimental::transform_reduce_by_adj_matrix_col_key_e(
+      handle,
+      graph_view,
+      cluster_v.begin(),
+      cluster_v.begin(),
+      cluster_v.begin(),
+      [] __device__(auto r, auto c, auto w, auto rv, auto cv) {
+        printf("transform reduce (%d,%d,%g,%d,%d)\n", (int)r, (int)c, (float)w, (int)rv, (int)cv);
```

Comment:
I see this transform reduce print statement 12 times, corresponding to the 12 edges in the graph.  This is as I expect.